event_driven_macros/src/expand.rs

```diff
@@ -86,40 +86,68 @@ pub fn expand(fsm: &mut Fsm) -> Result<TokenStream> {
             Some(ident_from_type(&t.from_state)?)
         };
         let command = ident_from_type(&t.command)?;
-        let event = ident_from_type(&t.event)?;
+        let event = if let Some(event) = &t.event {
+            Some(ident_from_type(event)?)
+        } else {
+            None
+        };
         let to_state = if let Some(to_state) = &t.to_state {
             Some(ident_from_type(to_state)?)
         } else {
             None
         };
 
         if let Some(from_state) = from_state {
-            let command_handler =
-                lowercase_ident(&format_ident!("for_{}_{}_{}", from_state, command, event));
+            if let Some(event) = event {
+                let command_handler =
+                    lowercase_ident(&format_ident!("for_{}_{}_{}", from_state, command, event));
+                command_matches.push(quote!(
+                    (#state_enum::#from_state(s), #command_enum::#command(c)) => {
+                        Self::#command_handler(s, c, se).map(|r| #event_enum::#event(r))
+                    }
+                ));
+            } else {
+                let command_handler =
+                    lowercase_ident(&format_ident!("for_{}_{}", from_state, command));
+                command_matches.push(quote!(
+                    (#state_enum::#from_state(s), #command_enum::#command(c)) => {
+                        Self::#command_handler(s, c, se)
```

Comment:
Should be: Self::#command_handler(s, c, se); None
?

Reply:
Whoops! Need tests for expand…